go/engine/login_provision.go

```diff
@@ -1063,65 +1063,65 @@ func (e *loginProvision) cleanup(m libkb.MetaContext) {
 	m.G().Logout()
 }
 
-func (e *loginProvision) verifyLocalStorage(m libkb.MetaContext) {
-	m.CDebugf("loginProvision: verifying local storage")
-	defer m.CDebugf("loginProvision: done verifying local storage")
-	normUsername := libkb.NewNormalizedUsername(e.username)
+var devtypeSortOrder = map[string]int{libkb.DeviceTypeMobile: 0, libkb.DeviceTypeDesktop: 1, libkb.DeviceTypePaper: 2}
+
+type partitionDeviceList []*libkb.Device
+
+func (p partitionDeviceList) Len() int {
+	return len(p)
+}
+
+func (p partitionDeviceList) Less(a, b int) bool {
+	if p[a].Type != p[b].Type {
+		return devtypeSortOrder[p[a].Type] < devtypeSortOrder[p[b].Type]
+	}
+	return *p[a].Description < *p[b].Description
+}
+
+func (p partitionDeviceList) Swap(a, b int) {
+	p[a], p[b] = p[b], p[a]
+}
+
+func verifyLocalStorage(m libkb.MetaContext, username string, uid keybase1.UID) {
+	m.CDebugf("verifying local storage")
+	defer m.CDebugf("done verifying local storage")
+	normUsername := libkb.NewNormalizedUsername(username)
 
 	// check config.json looks ok
-	e.verifyRegularFile(m, "config", m.G().Env.GetConfigFilename())
+	verifyRegularFile(m, "config", m.G().Env.GetConfigFilename())
 	cr := m.G().Env.GetConfig()
 	if cr.GetUsername() != normUsername {
-		m.CDebugf("loginProvision(verify): config username %q doesn't match engine username %q", cr.GetUsername(), normUsername)
+		m.CDebugf("config username %q doesn't match engine username %q", cr.GetUsername(), normUsername)
 	}
-	if cr.GetUID().NotEqual(e.arg.User.GetUID()) {
-		m.CDebugf("loginProvision(verify): config uid %q doesn't match engine uid %q", cr.GetUID(), e.arg.User.GetUID())
+	if cr.GetUID().NotEqual(uid) {
+		m.CDebugf("config uid %q doesn't match engine uid %q", cr.GetUID(), uid)
 	}
 
 	// check session.json is valid
-	e.verifyRegularFile(m, "session", m.G().Env.GetSessionFilename())
+	verifyRegularFile(m, "session", m.G().Env.GetSessionFilename())
```

Comment:
i'm surprised this still works, since i didn't think we still had this file.  maybe get rid of this?

Reply:
the check always fails here actually (we just log the errors) and i wasn't sure why that was the case. i'll drop the check and fail on the errors